runtime/runtime.go

```diff
@@ -75,25 +77,49 @@ func (c *Context) InitializeCodesAndPrograms() {
 	}
 }
 
+// Runnable can be run by a runner
+//
+// Runnable provides read-only data needed for execution
+type Runnable interface {
+	// Source returns the cadence script to be executed
+	Source() []byte
+	// Arguments returns arguments for this script
+	Arguments() [][]byte
+	// Authorizers returns a list address who has authorized this script
+	Authorizers() []Address
+	// ComputationLimit returns the max computation limit allowed while running
+	// Ramtin: (we might not need this to be passed and just be enforced in the Results)
+	ComputationLimit() uint64
+	// TODO: get rid of this decoding argument and merge it with arguments()
+	// DecodeArgument decodes a transaction argument against the given type.
+	DecodeArgument(argument []byte, argumentType cadence.Type) (cadence.Value, error)
```

Comment:
As discussed, maybe we can just pass decoded arguments directly, i.e. remove `DecodeArgument` and change `Arguments()` to `[]cadence.Value`